source/common/http/http2/codec_impl.cc

```diff
@@ -323,16 +322,16 @@ MetadataEncoder& ConnectionImpl::StreamImpl::getMetadataEncoder() {
 
 MetadataDecoder& ConnectionImpl::StreamImpl::getMetadataDecoder() {
   if (metadata_decoder_ == nullptr) {
-    auto cb = [this](std::unique_ptr<MetadataMap> metadata_map) {
-      this->onMetadataDecoded(std::move(metadata_map));
+    auto cb = [this](MetadataMapPtr metadata_map_ptr) {
```

Comment:
If you are going to end up moving, would prefer the signature be `MetadataMapPtr&&` (same elsewhere if applicable)